dbms/src/Functions/FunctionsString.cpp

```diff
@@ -5022,52 +5025,415 @@ class FunctionPosition : public IFunction
-        return static_cast<size_t>(UTF8::countCodePoints(data, idx) + 1);
+        size_t row_num = c0_col->size();
+
+        auto col_res = ColumnInt64::create(row_num, 0);
+        PaddedPODArray<Int64> & vec_res = col_res->getData();
+
+        if (c0_const && c1_string)
+        {
+            const String & c0_str = c0_const->getValue<String>();
+            if unlikely (c0_str.empty())
+            {
+                for (size_t i = 0; i < row_num; i++)
+                    vec_res[i] = 1;
+            }
+            else
+            {
+                constVector(c0_const->getValue<String>(), c1_string->getChars(), c1_string->getOffsets(), vec_res);
+            }
+        }
+        else if (c0_string && c1_string)
+        {
+            vectorVector(
+                c0_string->getChars(),
+                c0_string->getOffsets(),
+                c1_string->getChars(),
+                c1_string->getOffsets(),
+                vec_res);
+        }
+        else if (c0_string && c1_const)
+        {
+            vectorConst(c0_string->getChars(), c0_string->getOffsets(), c1_const->getValue<String>(), vec_res);
+        }
+        else
+            throw Exception(
+                "Illegal columns " + block.getByPosition(arguments[0]).column->getName() + " and "
+                    + block.getByPosition(arguments[1]).column->getName() + " of arguments of function " + getName(),
+                ErrorCodes::ILLEGAL_COLUMN);
+
+        block.getByPosition(result).column = std::move(col_res);
     }
+
+    void vectorVector(
+        const ColumnString::Chars_t & col0_data,
+        const ColumnString::Offsets & col0_offsets,
+        const ColumnString::Chars_t & col1_data,
+        const ColumnString::Offsets & col1_offsets,
+        PaddedPODArray<Int64> & res) const
+    {
+        size_t row_num = col0_offsets.size();
+        ColumnString::Offset prev_col0_str_offset = 0;
+        ColumnString::Offset prev_col1_str_offset = 0;
+
+        String col0_container;
+        String col1_container;
+
+        std::vector<size_t> lens;
+
+        for (size_t i = 0; i < row_num; i++)
+        {
+            size_t col0_str_len = col0_offsets[i] - prev_col0_str_offset - 1;
+            size_t col1_str_len = col1_offsets[i] - prev_col1_str_offset - 1;
+
+            if unlikely (col0_str_len == 0)
+            {
+                res[i] = 1;
+            }
+            else
+            {
+                StringRef col0_collation_str;
+
+                if (collator->isPaddingBinary())
+                    col0_collation_str = StringRef(&col0_data[prev_col0_str_offset], col0_str_len);
+                else
+                    col0_collation_str = collator->sortKeyNoTrim(
+                        reinterpret_cast<const char *>(&col0_data[prev_col0_str_offset]),
+                        col0_str_len,
+                        col0_container);
+
+                const StringRef & col1_collation_str = collator->convert(
```

Comment:
also add fast path if `collator->isPaddingBinary()` is true?

Reply:
> also add fast path if `collator->isPaddingBinary()` is true?

done